gabbi/tests/gabbits_intercept/utf8.txt

```diff
@@ -0,0 +1,222 @@
+<h1>Unicode Demo</h1>
+
+<p>Taken from <a
+href="http://www.cl.cam.ac.uk/~mgk25/ucs/examples/UTF-8-demo.txt">http://www.cl.cam.ac.uk/~mgk25/ucs/examples/UTF-8-demo.txt</a></p>
+
+<pre>
+
+$ENVIRON['HOME']
```

Comment:
It's always been a feature (but probably not well documented) and until you started using the method for response checking, was the main reason the data was being decoded from bytes to text: so the regular expressions could work. If the replace_template method wasn't being called there would be a nearly direct path to the data that was being read used as a raw body.

However, following the rule of thumb that we should be bytes at the io boundaries and text internally, the change to text and then back to bytes, is a good thing.

Make sense? And if so, do you concur?